src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/cache_tuning/FieldDataCacheDeciderITest.java

```diff
@@ -0,0 +1,255 @@
+                                        dimensionValues = {AllMetrics.GCType.Constants.HEAP_VALUE},
+                                        sum = 1000000.0, avg = 1000000.0, min = 1000000.0, max = 1000000.0)
+                        }),
+                @ATable(
+                        hostTag = {HostTag.ELECTED_MASTER},
+                        tuple = {
+                                @ATuple(
+                                        dimensionValues = {AllMetrics.GCType.Constants.HEAP_VALUE},
+                                        sum = 1000000.0, avg = 1000000.0, min = 1000000.0, max = 1000000.0),
+                                @ATuple(
+                                        dimensionValues = {AllMetrics.GCType.Constants.HEAP_VALUE},
+                                        sum = 1000000.0, avg = 1000000.0, min = 1000000.0, max = 1000000.0)
+                        })
+        })
+
+public class FieldDataCacheDeciderITest {
+    public static final String CACHE_TUNING_RESOURCES_DIR = Consts.INTEG_TESTS_SRC_DIR + "./tests/cache_tuning/resource/";
+    public static final String INDEX_NAME = "MockIndex";
+    public static final String SHARD_ID = "1";
+
+    // Test CacheDecider for ModifyCacheAction (field data cache).
+    // The cache decider should emit modify cache size action as field data rca is unhealthy.
+    @Test
+    @AExpect(
+            what = AExpect.Type.DB_QUERY,
+            on = HostTag.ELECTED_MASTER,
+            validator = FieldDataCacheDeciderValidator.class,
+            forRca = PersistedAction.class,
+            timeoutSeconds = 1000)
+    @AErrorPatternIgnored(
+            pattern = "AggregateMetric:gather()",
+            reason = "CPU metrics are expected to be missing in this integ test")
+    @AErrorPatternIgnored(
+            pattern = "Metric:gather()",
+            reason = "Metrics are expected to be missing in this integ test")
+    @AErrorPatternIgnored(
+            pattern = "NodeConfigCacheReaderUtil",
+            reason = "Node Config Cache are expected to be missing in this integ test.")
+    @AErrorPatternIgnored(
+            pattern = "CacheUtil:getCacheMaxSize()",
+            reason = "Node Config Cache is expected to be missing during startup.")
+    @AErrorPatternIgnored(
+            pattern = "SubscribeResponseHandler:onError()",
+            reason = "A unit test expressly calls SubscribeResponseHandler#onError, which writes an error log")
+    @AErrorPatternIgnored(
+            pattern = "SQLParsingUtil:readDataFromSqlResult()",
+            reason = "Old gen metrics is expected to be missing in this integ test.")
+    @AErrorPatternIgnored(
+            pattern = "HighHeapUsageOldGenRca:operate()",
+            reason = "Old gen rca is expected to be missing in this integ test.")
+    @AErrorPatternIgnored(
+            pattern = "ModifyCacheMaxSizeAction:build()",
+            reason = "Node Config Cache metrics is expected to be missing during startup")
```

Comment:
Can we make the Node Config cache available in IT?

Reply:
All cache metrics are available in node config cache. Removing these.

Reply:
All cache metrics are available in node config cache. But during shutdown, node config cache gets cleared before stopping the deciders and hence we are seeing this error. 
Re-running the tests, makes the tests pass. Adding this to avoid flaky tests.